src/main/kotlin/no/nav/familie/ef/sak/api/gui/VedtaksbrevController.kt

```diff
@@ -16,20 +17,25 @@ import java.util.*
 class VedtaksbrevController(private val brevService: VedtaksbrevService,
                             private val tilgangService: TilgangService) {
 
+    private val logger = LoggerFactory.getLogger(this.javaClass)
     @PostMapping("/{behandlingId}/{brevMal}")
-    fun forhåndsvisBrevV2(@PathVariable behandlingId: UUID,
-                          @PathVariable brevMal: String,
-                          @RequestBody brevRequest: JsonNode): Ressurs<ByteArray> {
+    fun lagBrev(@PathVariable behandlingId: UUID,
+                @PathVariable brevMal: String,
+                @RequestBody brevRequest: JsonNode): Ressurs<ByteArray> {
         tilgangService.validerTilgangTilBehandling(behandlingId)
-
-        val respons = brevService.forhåndsvisBrev(behandlingId, brevRequest, brevMal)
-
-        return Ressurs.success(respons)
+        val data = brevService.lagSaksbehandlerBrev(behandlingId, brevRequest, brevMal)
+        return Ressurs.success(data)
     }
 
     @GetMapping("/{behandlingId}")
     fun hentBrev(@PathVariable behandlingId: UUID): Ressurs<ByteArray> {
         tilgangService.validerTilgangTilBehandling(behandlingId)
-        return  Ressurs.success(brevService.hentBrev(behandlingId).utkastPdf.bytes)
+        val lagBeslutterBrev = brevService.lagBeslutterBrev(behandlingId)
```

Comment:
Metoden her heter hentBrev, men det den gjør er å opprette beslutterbrev?

Hvis man endrer lagBeslutterBrev til noe slik kanskje, så unngår man denne try catch med nullpointer. 
```
val besluttervedtaksbrevMedPdf = besluttervedtaksbrev.copy(beslutterPdf = beslutterPdf)
brevRepository.update(besluttervedtaksbrevMedPdf)
return beslutterPdf
```

Hvis ikke så synes jeg det er ryddigere å bruke
```
        val beslutterPdf = brevService.lagBeslutterBrev(behandlingId).beslutterPdf
        return if (beslutterPdf != null) {
            Ressurs.success(beslutterPdf.bytes)
        } else {
            logger.error("Pdf finnes ikke for behandling=$behandlingId.")
            Ressurs.failure("Pdf for beslutter kunne ikke genereres")
        }
```
Men slik koden er nå skal det aldri kunne skje att den skal returnere null for beslutterPdf, så føles litt unødvendig med hele denne sjekken :P

Reply:
hentBrev burde kanskje endre navn til lagBeslutterbrev i controller også? Samme med den andre -lagBrev => lagSaksbehandlerbrev?

Reply:
Vi gikk for "løsning to" her. Enig i at det er ryddigere.

Reply:
Jeg vet ikke, jeg synes fortsatt denne blir litt rar. Servicen trenger ju bare å returnere beslutterPdf, som ikke kan være null. Så unngår man en unødvendig nullsjekk som aldri kommer skje?